File: javaapp/src/main/java/com/example/javaapp/PersonsAdapter.java
package com.example.javaapp;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.javaapp.data.Person;

import java.util.ArrayList;
import java.util.List;

public class PersonsAdapter extends RecyclerView.Adapter<PersonsAdapter.PersonsViewHolder> {
    private List<Person> dataSet;
    private ItemClickListener itemClickListener;

    public static class PersonsViewHolder extends RecyclerView.ViewHolder {
        private TextView tvName;
        private TextView tvAge;
        private TextView tvGender;

        public PersonsViewHolder(@NonNull View itemView) {
            super(itemView);
            tvName = itemView.findViewById(R.id.list_tv_name);
            tvAge = itemView.findViewById(R.id.list_tv_age);
            tvGender = itemView.findViewById(R.id.list_tv_gender);
        }

        protected void bindInfo(Person data) {
            tvName.setText(data.getFullName());
            tvAge.setText(Integer.toString(data.getAge()));
            tvGender.setText(data.getGender());
        }
    }

    public PersonsAdapter(List<Person> dataSet) {
        this.dataSet = new ArrayList<>(dataSet);
    }

    @NonNull
    @Override
    public PersonsViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        // 리스트 아이템 UI를 정의한 뷰 생성
        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.list_person_item, parent, false);
        return new PersonsViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull PersonsViewHolder holder, int position) {
        holder.bindInfo(dataSet.get(position));
        holder.itemView.setOnClickListener(view -> {
            itemClickListener.onClick(dataSet.get(position));
        });
    }

    @Override
    public int getItemCount() {
        return dataSet.size();
    }

    public void setItemClickListener(ItemClickListener itemClickListener) {
        this.itemClickListener = itemClickListener;
    }

    interface ItemClickListener {
        void onClick(Person person);
    }
}

File: app/src/main/java/com/example/androidcs/PersonsAdapter.kt
package com.example.androidcs

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.example.androidcs.data.Person
import com.example.androidcs.data.getFullName

class PersonsAdapter(private val dataSet: List<Person>) :
    RecyclerView.Adapter<PersonsAdapter.PersonsViewHolder>() {
    private lateinit var itemClickListener: ItemClickListener

    inner class PersonsViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        private val tvName: TextView
        private val tvAge: TextView
        private val tvGender: TextView

        init {
            tvName = view.findViewById(R.id.list_tv_name)
            tvAge = view.findViewById(R.id.list_tv_age)
            tvGender = view.findViewById(R.id.list_tv_gender)
        }

        fun bindInfo(data: Person) {
            tvName.text = data.getFullName()
            tvAge.text = data.age.toString()
            tvGender.text = data.gender
        }
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): PersonsViewHolder {
        // 리스트 아이템 UI를 정의한 뷰 생성
        val view = LayoutInflater.from(parent.context)
            .inflate(R.layout.list_person_item, parent, false)
        return PersonsViewHolder(view)
    }

    override fun onBindViewHolder(holder: PersonsViewHolder, position: Int) {
        holder.apply {
            bindInfo(dataSet[position])
            itemView.setOnClickListener {
                itemClickListener.onClick(dataSet[position])
            }
        }
    }

    override fun getItemCount(): Int = dataSet.size

    fun setItemClickListener(itemClickListener: ItemClickListener) {
        this.itemClickListener = itemClickListener
    }

    interface ItemClickListener {
        fun onClick(person: Person)
    }
}